packages/beacon-node/src/chain/validation/blobSidecar.ts

```diff
@@ -0,0 +1,154 @@
+): void {
+  // TODO: freetheblobs
+  if (!verifyKzgCommitmentsAgainstTransactions(transactions, expectedKzgCommitments)) {
+    throw new Error("Invalid block transactions and commitments");
+  }
+
+  // assert len(expected_kzg_commitments) == len(blobs)
+  if (expectedKzgCommitments.length !== blobSidecars.length) {
+    throw new Error(
+      `blobSidecars length to commitments length mismatch. Blob length: ${blobSidecars.length}, Expected commitments length ${expectedKzgCommitments.length}`
+    );
+  }
+
+  // No need to verify the aggregate proof of zero blobs. Also c-kzg throws.
+  // https://github.com/dankrad/c-kzg/pull/12/files#r1025851956
+  if (blobSidecars.length > 0) {
+    // Verify the blob slot and root matches
+    const blobs = [];
+    const proofs = [];
+    for (let index = 0; index < blobSidecars.length; index++) {
+      const blobSidecar = blobSidecars[index];
+      if (
+        blobSidecar.slot !== blockSlot ||
+        !byteArrayEquals(blobSidecar.blockRoot, blockRoot) ||
+        blobSidecar.index !== index ||
+        !byteArrayEquals(expectedKzgCommitments[index], blobSidecar.kzgCommitment)
+      ) {
+        throw new Error(
+          `Invalid blob with slot=${blobSidecar.slot} blockRoot=${toHex(blockRoot)} index=${
+            blobSidecar.index
+          } for the block root=${toHex(blockRoot)} slot=${blockSlot} index=${index}`
+        );
+      }
+      blobs.push(blobSidecar.blob);
+      proofs.push(blobSidecar.kzgProof);
+    }
+    validateBlobsAndProofs(expectedKzgCommitments, blobs, proofs);
+  }
+}
+
+function validateBlobsAndProofs(
+  expectedKzgCommitments: deneb.BlobKzgCommitments,
+  blobs: deneb.Blobs,
+  proofs: deneb.KZGProofs
+): void {
+  // assert verify_aggregate_kzg_proof(blobs, expected_kzg_commitments, kzg_aggregated_proof)
+  let isProofValid: boolean;
+  try {
+    isProofValid = ckzg.verifyBlobKzgProofBatch(blobs, expectedKzgCommitments, proofs);
+  } catch (e) {
+    (e as Error).message = `Error on verifyBlobKzgProofBatch: ${(e as Error).message}`;
+    throw e;
+  }
+  if (!isProofValid) {
+    throw Error("Invalid verifyBlobKzgProofBatch");
+  }
+}
+
+/**
+ * From https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-2.5
+ * KeyValidate = valid, non-identity point that is in the correct subgroup
+ */
+function blsKeyValidate(g1Point: Uint8Array): boolean {
+  try {
+    bls.PublicKey.fromBytes(g1Point, CoordType.jacobian, true);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+/**
+ * ```
+ * Blob = new ByteVectorType(BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB);
+ * ```
+ * Check that each FIELD_ELEMENT as a uint256 < BLS_MODULUS
+ */
+function blobIsValidRange(blob: deneb.Blob): boolean {
+  for (let i = 0; i < FIELD_ELEMENTS_PER_BLOB; i++) {
+    const fieldElement = blob.subarray(i * BYTES_PER_FIELD_ELEMENT, (i + 1) * BYTES_PER_FIELD_ELEMENT);
+    const fieldElementBN = bytesToBigInt(fieldElement, "be");
```

Comment:
We use little endian internally in the cryptography layer, so its a bit suprising that we are using big endian to convert it to an integer here. 

Moreover, this seems like an abstraction leak and perhaps the cryptography code could expose a `isValidBlob` method to cover this

Reply:
yes that would be useful, but i guess verifyBlobKzgProof will verify all three blob, kzg and proof, let me know if that assumption is not correct